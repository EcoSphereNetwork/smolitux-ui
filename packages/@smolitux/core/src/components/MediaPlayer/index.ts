export { MediaPlayer } from './MediaPlayer';
export type { MediaPlayerProps, MediaType, MediaSource, MediaTrack } from './MediaPlayer';

// Fur Abwartskompatibilitat mit dem bestehenden Export
export { MediaPlayer as default } from './MediaPlayer';