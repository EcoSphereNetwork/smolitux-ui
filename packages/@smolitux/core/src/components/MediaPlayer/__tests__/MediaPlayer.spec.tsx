import React from 'react';
import { render } from '@testing-library/react';
import { MediaPlayer } from '../MediaPlayer';

// Mock für ThemeProvider
jest.mock('@smolitux/theme', () => ({
  useTheme: jest.fn(() => ({ themeMode: 'light' })),
}));

// Mock für HTMLMediaElement
window.HTMLMediaElement.prototype.load = jest.fn();
window.HTMLMediaElement.prototype.play = jest.fn(() => Promise.resolve());
window.HTMLMediaElement.prototype.pause = jest.fn();

describe('MediaPlayer Snapshots', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders default video player correctly', () => {
    const { asFragment } = render(<MediaPlayer src="test-video.mp4" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders audio player correctly', () => {
    const { asFragment } = render(<MediaPlayer src="test-audio.mp3" type="audio" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with multiple sources correctly', () => {
    const sources = [
      { src: 'test-video.mp4', type: 'video/mp4' },
      { src: 'test-video.webm', type: 'video/webm' }
    ];
    
    const { asFragment } = render(<MediaPlayer sources={sources} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with tracks correctly', () => {
    const tracks = [
      { src: 'subtitles-en.vtt', srclang: 'en', label: 'English', kind: 'subtitles' as const },
      { src: 'subtitles-de.vtt', srclang: 'de', label: 'Deutsch', kind: 'subtitles' as const }
    ];
    
    const { asFragment } = render(<MediaPlayer src="test-video.mp4" tracks={tracks} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with poster image correctly', () => {
    const { asFragment } = render(<MediaPlayer src="test-video.mp4" poster="poster.jpg" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with autoplay correctly', () => {
    const { asFragment } = render(<MediaPlayer src="test-video.mp4" autoPlay />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with loop correctly', () => {
    const { asFragment } = render(<MediaPlayer src="test-video.mp4" loop />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with controls hidden correctly', () => {
    const { asFragment } = render(<MediaPlayer src="test-video.mp4" hideControls />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with custom width and height correctly', () => {
    const { asFragment } = render(<MediaPlayer src="test-video.mp4" width={800} height={600} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with custom className correctly', () => {
    const { asFragment } = render(<MediaPlayer src="test-video.mp4" className="custom-media-player" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with different control variants correctly', () => {
    const variants = ['default', 'minimal', 'transparent', 'floating'];
    
    const fragments = variants.map(variant => {
      const { asFragment } = render(<MediaPlayer src="test-video.mp4" controlsVariant={variant as any} />);
      return { variant, fragment: asFragment() };
    });
    
    fragments.forEach(({ variant, fragment }) => {
      expect(fragment).toMatchSnapshot(`MediaPlayer with controls variant ${variant}`);
    });
  });

  it('renders with different aspect ratios correctly', () => {
    const aspectRatios = ['16:9', '4:3', '1:1', '21:9'];
    
    const fragments = aspectRatios.map(aspectRatio => {
      const { asFragment } = render(<MediaPlayer src="test-video.mp4" aspectRatio={aspectRatio as any} />);
      return { aspectRatio, fragment: asFragment() };
    });
    
    fragments.forEach(({ aspectRatio, fragment }) => {
      expect(fragment).toMatchSnapshot(`MediaPlayer with aspect ratio ${aspectRatio}`);
    });
  });

  it('renders with different quality options correctly', () => {
    const sources = [
      { src: 'test-video-hd.mp4', type: 'video/mp4', quality: 'HD', bitrate: 5000 },
      { src: 'test-video-sd.mp4', type: 'video/mp4', quality: 'SD', bitrate: 2000 },
      { src: 'test-video-low.mp4', type: 'video/mp4', quality: 'Low', bitrate: 800 }
    ];
    
    const { asFragment } = render(<MediaPlayer sources={sources} showQualitySelector />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with playback speed controls correctly', () => {
    const { asFragment } = render(<MediaPlayer src="test-video.mp4" showSpeedControls />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with all features enabled correctly', () => {
    const sources = [
      { src: 'test-video-hd.mp4', type: 'video/mp4', quality: 'HD', bitrate: 5000 },
      { src: 'test-video-sd.mp4', type: 'video/mp4', quality: 'SD', bitrate: 2000 }
    ];
    
    const tracks = [
      { src: 'subtitles-en.vtt', srclang: 'en', label: 'English', kind: 'subtitles' as const },
      { src: 'subtitles-de.vtt', srclang: 'de', label: 'Deutsch', kind: 'subtitles' as const }
    ];
    
    const { asFragment } = render(
      <MediaPlayer 
        sources={sources}
        tracks={tracks}
        poster="poster.jpg"
        autoPlay
        loop
        width={800}
        height={600}
        showQualitySelector
        showSpeedControls
        aspectRatio="16:9"
        controlsVariant="floating"
        className="custom-media-player"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});