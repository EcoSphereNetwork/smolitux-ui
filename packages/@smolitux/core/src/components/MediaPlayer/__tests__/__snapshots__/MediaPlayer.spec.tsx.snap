// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaPlayer Snapshots renders audio player correctly 1`] = `
<DocumentFragment>
  <div
    class="relative "
    data-testid="media-player"
  >
    <audio
      class="w-full"
      controls=""
      data-testid="audio-element"
      src="test-audio.mp3"
    />
    <div
      class="absolute bottom-0 left-0 right-0"
      data-testid="media-controls"
    >
      <div
        class="h-1 bg-gray-600 w-full"
        data-testid="progress-bar"
      >
        <div
          class="h-full bg-primary-500"
          style="width: 0%;"
        />
      </div>
      <div
        class="flex items-center p-2"
      >
        <button
          aria-label="Play"
          class="mr-2"
        >
          ▶
        </button>
        <button
          aria-label="Pause"
          class="mr-2"
        >
          ⏸
        </button>
        <button
          aria-label="Mute"
          class="mr-2"
        >
          🔇
        </button>
        <button
          aria-label="Unmute"
          class="mr-2 hidden"
        >
          🔊
        </button>
        <input
          aria-label="Volume"
          class="w-24 mr-2"
          max="1"
          min="0"
          step="0.1"
          type="range"
          value="1"
        />
        <button
          aria-label="Enter fullscreen"
        >
          ⛶
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MediaPlayer Snapshots renders default video player correctly 1`] = `
<DocumentFragment>
  <div
    class="relative "
    data-testid="media-player"
  >
    <video
      class="w-full h-full"
      controls=""
      data-testid="video-element"
      src="test-video.mp4"
    />
    <div
      class="absolute bottom-0 left-0 right-0"
      data-testid="media-controls"
    >
      <div
        class="h-1 bg-gray-600 w-full"
        data-testid="progress-bar"
      >
        <div
          class="h-full bg-primary-500"
          style="width: 0%;"
        />
      </div>
      <div
        class="flex items-center p-2"
      >
        <button
          aria-label="Play"
          class="mr-2"
        >
          ▶
        </button>
        <button
          aria-label="Pause"
          class="mr-2"
        >
          ⏸
        </button>
        <button
          aria-label="Mute"
          class="mr-2"
        >
          🔇
        </button>
        <button
          aria-label="Unmute"
          class="mr-2 hidden"
        >
          🔊
        </button>
        <input
          aria-label="Volume"
          class="w-24 mr-2"
          max="1"
          min="0"
          step="0.1"
          type="range"
          value="1"
        />
        <button
          aria-label="Enter fullscreen"
        >
          ⛶
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MediaPlayer Snapshots renders with all features enabled correctly 1`] = `
<DocumentFragment>
  <div
    aspectratio="16:9"
    class="relative custom-media-player"
    controlsvariant="floating"
    data-testid="media-player"
    style="width: 800px; height: 600px;"
  >
    <video
      autoplay=""
      class="w-full h-full"
      controls=""
      data-testid="video-element"
      height="600"
      loop=""
      poster="poster.jpg"
      width="800"
    >
      <source
        data-testid="source-element"
        src="test-video-hd.mp4"
        type="video/mp4"
      />
      <source
        data-testid="source-element"
        src="test-video-sd.mp4"
        type="video/mp4"
      />
      <track
        data-testid="track-element"
        kind="subtitles"
        label="English"
        src="subtitles-en.vtt"
        srclang="en"
      />
      <track
        data-testid="track-element"
        kind="subtitles"
        label="Deutsch"
        src="subtitles-de.vtt"
        srclang="de"
      />
    </video>
    <div
      class="absolute bottom-0 left-0 right-0"
      data-testid="media-controls"
    >
      <div
        class="h-1 bg-gray-600 w-full"
        data-testid="progress-bar"
      >
        <div
          class="h-full bg-primary-500"
          style="width: 0%;"
        />
      </div>
      <div
        class="flex items-center p-2"
      >
        <button
          aria-label="Play"
          class="mr-2"
        >
          ▶
        </button>
        <button
          aria-label="Pause"
          class="mr-2"
        >
          ⏸
        </button>
        <button
          aria-label="Mute"
          class="mr-2"
        >
          🔇
        </button>
        <button
          aria-label="Unmute"
          class="mr-2 hidden"
        >
          🔊
        </button>
        <input
          aria-label="Volume"
          class="w-24 mr-2"
          max="1"
          min="0"
          step="0.1"
          type="range"
          value="1"
        />
        <button
          aria-label="Enter fullscreen"
        >
          ⛶
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MediaPlayer Snapshots renders with autoplay correctly 1`] = `
<DocumentFragment>
  <div
    class="relative "
    data-testid="media-player"
  >
    <video
      autoplay=""
      class="w-full h-full"
      controls=""
      data-testid="video-element"
      src="test-video.mp4"
    />
    <div
      class="absolute bottom-0 left-0 right-0"
      data-testid="media-controls"
    >
      <div
        class="h-1 bg-gray-600 w-full"
        data-testid="progress-bar"
      >
        <div
          class="h-full bg-primary-500"
          style="width: 0%;"
        />
      </div>
      <div
        class="flex items-center p-2"
      >
        <button
          aria-label="Play"
          class="mr-2"
        >
          ▶
        </button>
        <button
          aria-label="Pause"
          class="mr-2"
        >
          ⏸
        </button>
        <button
          aria-label="Mute"
          class="mr-2"
        >
          🔇
        </button>
        <button
          aria-label="Unmute"
          class="mr-2 hidden"
        >
          🔊
        </button>
        <input
          aria-label="Volume"
          class="w-24 mr-2"
          max="1"
          min="0"
          step="0.1"
          type="range"
          value="1"
        />
        <button
          aria-label="Enter fullscreen"
        >
          ⛶
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MediaPlayer Snapshots renders with controls hidden correctly 1`] = `
<DocumentFragment>
  <div
    class="relative "
    data-testid="media-player"
  >
    <video
      class="w-full h-full"
      controls=""
      data-testid="video-element"
      src="test-video.mp4"
    />
  </div>
</DocumentFragment>
`;

exports[`MediaPlayer Snapshots renders with custom className correctly 1`] = `
<DocumentFragment>
  <div
    class="relative custom-media-player"
    data-testid="media-player"
  >
    <video
      class="w-full h-full"
      controls=""
      data-testid="video-element"
      src="test-video.mp4"
    />
    <div
      class="absolute bottom-0 left-0 right-0"
      data-testid="media-controls"
    >
      <div
        class="h-1 bg-gray-600 w-full"
        data-testid="progress-bar"
      >
        <div
          class="h-full bg-primary-500"
          style="width: 0%;"
        />
      </div>
      <div
        class="flex items-center p-2"
      >
        <button
          aria-label="Play"
          class="mr-2"
        >
          ▶
        </button>
        <button
          aria-label="Pause"
          class="mr-2"
        >
          ⏸
        </button>
        <button
          aria-label="Mute"
          class="mr-2"
        >
          🔇
        </button>
        <button
          aria-label="Unmute"
          class="mr-2 hidden"
        >
          🔊
        </button>
        <input
          aria-label="Volume"
          class="w-24 mr-2"
          max="1"
          min="0"
          step="0.1"
          type="range"
          value="1"
        />
        <button
          aria-label="Enter fullscreen"
        >
          ⛶
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MediaPlayer Snapshots renders with custom width and height correctly 1`] = `
<DocumentFragment>
  <div
    class="relative "
    data-testid="media-player"
    style="width: 800px; height: 600px;"
  >
    <video
      class="w-full h-full"
      controls=""
      data-testid="video-element"
      height="600"
      src="test-video.mp4"
      width="800"
    />
    <div
      class="absolute bottom-0 left-0 right-0"
      data-testid="media-controls"
    >
      <div
        class="h-1 bg-gray-600 w-full"
        data-testid="progress-bar"
      >
        <div
          class="h-full bg-primary-500"
          style="width: 0%;"
        />
      </div>
      <div
        class="flex items-center p-2"
      >
        <button
          aria-label="Play"
          class="mr-2"
        >
          ▶
        </button>
        <button
          aria-label="Pause"
          class="mr-2"
        >
          ⏸
        </button>
        <button
          aria-label="Mute"
          class="mr-2"
        >
          🔇
        </button>
        <button
          aria-label="Unmute"
          class="mr-2 hidden"
        >
          🔊
        </button>
        <input
          aria-label="Volume"
          class="w-24 mr-2"
          max="1"
          min="0"
          step="0.1"
          type="range"
          value="1"
        />
        <button
          aria-label="Enter fullscreen"
        >
          ⛶
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MediaPlayer Snapshots renders with different aspect ratios correctly: MediaPlayer with aspect ratio 1:1 1`] = `
<DocumentFragment>
  <div
    aspectratio="1:1"
    class="relative "
    data-testid="media-player"
  >
    <video
      class="w-full h-full"
      controls=""
      data-testid="video-element"
      src="test-video.mp4"
    />
    <div
      class="absolute bottom-0 left-0 right-0"
      data-testid="media-controls"
    >
      <div
        class="h-1 bg-gray-600 w-full"
        data-testid="progress-bar"
      >
        <div
          class="h-full bg-primary-500"
          style="width: 0%;"
        />
      </div>
      <div
        class="flex items-center p-2"
      >
        <button
          aria-label="Play"
          class="mr-2"
        >
          ▶
        </button>
        <button
          aria-label="Pause"
          class="mr-2"
        >
          ⏸
        </button>
        <button
          aria-label="Mute"
          class="mr-2"
        >
          🔇
        </button>
        <button
          aria-label="Unmute"
          class="mr-2 hidden"
        >
          🔊
        </button>
        <input
          aria-label="Volume"
          class="w-24 mr-2"
          max="1"
          min="0"
          step="0.1"
          type="range"
          value="1"
        />
        <button
          aria-label="Enter fullscreen"
        >
          ⛶
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MediaPlayer Snapshots renders with different aspect ratios correctly: MediaPlayer with aspect ratio 4:3 1`] = `
<DocumentFragment>
  <div
    aspectratio="4:3"
    class="relative "
    data-testid="media-player"
  >
    <video
      class="w-full h-full"
      controls=""
      data-testid="video-element"
      src="test-video.mp4"
    />
    <div
      class="absolute bottom-0 left-0 right-0"
      data-testid="media-controls"
    >
      <div
        class="h-1 bg-gray-600 w-full"
        data-testid="progress-bar"
      >
        <div
          class="h-full bg-primary-500"
          style="width: 0%;"
        />
      </div>
      <div
        class="flex items-center p-2"
      >
        <button
          aria-label="Play"
          class="mr-2"
        >
          ▶
        </button>
        <button
          aria-label="Pause"
          class="mr-2"
        >
          ⏸
        </button>
        <button
          aria-label="Mute"
          class="mr-2"
        >
          🔇
        </button>
        <button
          aria-label="Unmute"
          class="mr-2 hidden"
        >
          🔊
        </button>
        <input
          aria-label="Volume"
          class="w-24 mr-2"
          max="1"
          min="0"
          step="0.1"
          type="range"
          value="1"
        />
        <button
          aria-label="Enter fullscreen"
        >
          ⛶
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MediaPlayer Snapshots renders with different aspect ratios correctly: MediaPlayer with aspect ratio 16:9 1`] = `
<DocumentFragment>
  <div
    aspectratio="16:9"
    class="relative "
    data-testid="media-player"
  >
    <video
      class="w-full h-full"
      controls=""
      data-testid="video-element"
      src="test-video.mp4"
    />
    <div
      class="absolute bottom-0 left-0 right-0"
      data-testid="media-controls"
    >
      <div
        class="h-1 bg-gray-600 w-full"
        data-testid="progress-bar"
      >
        <div
          class="h-full bg-primary-500"
          style="width: 0%;"
        />
      </div>
      <div
        class="flex items-center p-2"
      >
        <button
          aria-label="Play"
          class="mr-2"
        >
          ▶
        </button>
        <button
          aria-label="Pause"
          class="mr-2"
        >
          ⏸
        </button>
        <button
          aria-label="Mute"
          class="mr-2"
        >
          🔇
        </button>
        <button
          aria-label="Unmute"
          class="mr-2 hidden"
        >
          🔊
        </button>
        <input
          aria-label="Volume"
          class="w-24 mr-2"
          max="1"
          min="0"
          step="0.1"
          type="range"
          value="1"
        />
        <button
          aria-label="Enter fullscreen"
        >
          ⛶
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MediaPlayer Snapshots renders with different aspect ratios correctly: MediaPlayer with aspect ratio 21:9 1`] = `
<DocumentFragment>
  <div
    aspectratio="21:9"
    class="relative "
    data-testid="media-player"
  >
    <video
      class="w-full h-full"
      controls=""
      data-testid="video-element"
      src="test-video.mp4"
    />
    <div
      class="absolute bottom-0 left-0 right-0"
      data-testid="media-controls"
    >
      <div
        class="h-1 bg-gray-600 w-full"
        data-testid="progress-bar"
      >
        <div
          class="h-full bg-primary-500"
          style="width: 0%;"
        />
      </div>
      <div
        class="flex items-center p-2"
      >
        <button
          aria-label="Play"
          class="mr-2"
        >
          ▶
        </button>
        <button
          aria-label="Pause"
          class="mr-2"
        >
          ⏸
        </button>
        <button
          aria-label="Mute"
          class="mr-2"
        >
          🔇
        </button>
        <button
          aria-label="Unmute"
          class="mr-2 hidden"
        >
          🔊
        </button>
        <input
          aria-label="Volume"
          class="w-24 mr-2"
          max="1"
          min="0"
          step="0.1"
          type="range"
          value="1"
        />
        <button
          aria-label="Enter fullscreen"
        >
          ⛶
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MediaPlayer Snapshots renders with different control variants correctly: MediaPlayer with controls variant default 1`] = `
<DocumentFragment>
  <div
    class="relative "
    controlsvariant="default"
    data-testid="media-player"
  >
    <video
      class="w-full h-full"
      controls=""
      data-testid="video-element"
      src="test-video.mp4"
    />
    <div
      class="absolute bottom-0 left-0 right-0"
      data-testid="media-controls"
    >
      <div
        class="h-1 bg-gray-600 w-full"
        data-testid="progress-bar"
      >
        <div
          class="h-full bg-primary-500"
          style="width: 0%;"
        />
      </div>
      <div
        class="flex items-center p-2"
      >
        <button
          aria-label="Play"
          class="mr-2"
        >
          ▶
        </button>
        <button
          aria-label="Pause"
          class="mr-2"
        >
          ⏸
        </button>
        <button
          aria-label="Mute"
          class="mr-2"
        >
          🔇
        </button>
        <button
          aria-label="Unmute"
          class="mr-2 hidden"
        >
          🔊
        </button>
        <input
          aria-label="Volume"
          class="w-24 mr-2"
          max="1"
          min="0"
          step="0.1"
          type="range"
          value="1"
        />
        <button
          aria-label="Enter fullscreen"
        >
          ⛶
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MediaPlayer Snapshots renders with different control variants correctly: MediaPlayer with controls variant floating 1`] = `
<DocumentFragment>
  <div
    class="relative "
    controlsvariant="floating"
    data-testid="media-player"
  >
    <video
      class="w-full h-full"
      controls=""
      data-testid="video-element"
      src="test-video.mp4"
    />
    <div
      class="absolute bottom-0 left-0 right-0"
      data-testid="media-controls"
    >
      <div
        class="h-1 bg-gray-600 w-full"
        data-testid="progress-bar"
      >
        <div
          class="h-full bg-primary-500"
          style="width: 0%;"
        />
      </div>
      <div
        class="flex items-center p-2"
      >
        <button
          aria-label="Play"
          class="mr-2"
        >
          ▶
        </button>
        <button
          aria-label="Pause"
          class="mr-2"
        >
          ⏸
        </button>
        <button
          aria-label="Mute"
          class="mr-2"
        >
          🔇
        </button>
        <button
          aria-label="Unmute"
          class="mr-2 hidden"
        >
          🔊
        </button>
        <input
          aria-label="Volume"
          class="w-24 mr-2"
          max="1"
          min="0"
          step="0.1"
          type="range"
          value="1"
        />
        <button
          aria-label="Enter fullscreen"
        >
          ⛶
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MediaPlayer Snapshots renders with different control variants correctly: MediaPlayer with controls variant minimal 1`] = `
<DocumentFragment>
  <div
    class="relative "
    controlsvariant="minimal"
    data-testid="media-player"
  >
    <video
      class="w-full h-full"
      controls=""
      data-testid="video-element"
      src="test-video.mp4"
    />
    <div
      class="absolute bottom-0 left-0 right-0"
      data-testid="media-controls"
    >
      <div
        class="h-1 bg-gray-600 w-full"
        data-testid="progress-bar"
      >
        <div
          class="h-full bg-primary-500"
          style="width: 0%;"
        />
      </div>
      <div
        class="flex items-center p-2"
      >
        <button
          aria-label="Play"
          class="mr-2"
        >
          ▶
        </button>
        <button
          aria-label="Pause"
          class="mr-2"
        >
          ⏸
        </button>
        <button
          aria-label="Mute"
          class="mr-2"
        >
          🔇
        </button>
        <button
          aria-label="Unmute"
          class="mr-2 hidden"
        >
          🔊
        </button>
        <input
          aria-label="Volume"
          class="w-24 mr-2"
          max="1"
          min="0"
          step="0.1"
          type="range"
          value="1"
        />
        <button
          aria-label="Enter fullscreen"
        >
          ⛶
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MediaPlayer Snapshots renders with different control variants correctly: MediaPlayer with controls variant transparent 1`] = `
<DocumentFragment>
  <div
    class="relative "
    controlsvariant="transparent"
    data-testid="media-player"
  >
    <video
      class="w-full h-full"
      controls=""
      data-testid="video-element"
      src="test-video.mp4"
    />
    <div
      class="absolute bottom-0 left-0 right-0"
      data-testid="media-controls"
    >
      <div
        class="h-1 bg-gray-600 w-full"
        data-testid="progress-bar"
      >
        <div
          class="h-full bg-primary-500"
          style="width: 0%;"
        />
      </div>
      <div
        class="flex items-center p-2"
      >
        <button
          aria-label="Play"
          class="mr-2"
        >
          ▶
        </button>
        <button
          aria-label="Pause"
          class="mr-2"
        >
          ⏸
        </button>
        <button
          aria-label="Mute"
          class="mr-2"
        >
          🔇
        </button>
        <button
          aria-label="Unmute"
          class="mr-2 hidden"
        >
          🔊
        </button>
        <input
          aria-label="Volume"
          class="w-24 mr-2"
          max="1"
          min="0"
          step="0.1"
          type="range"
          value="1"
        />
        <button
          aria-label="Enter fullscreen"
        >
          ⛶
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MediaPlayer Snapshots renders with different quality options correctly 1`] = `
<DocumentFragment>
  <div
    class="relative "
    data-testid="media-player"
  >
    <video
      class="w-full h-full"
      controls=""
      data-testid="video-element"
    >
      <source
        data-testid="source-element"
        src="test-video-hd.mp4"
        type="video/mp4"
      />
      <source
        data-testid="source-element"
        src="test-video-sd.mp4"
        type="video/mp4"
      />
      <source
        data-testid="source-element"
        src="test-video-low.mp4"
        type="video/mp4"
      />
    </video>
    <div
      class="absolute bottom-0 left-0 right-0"
      data-testid="media-controls"
    >
      <div
        class="h-1 bg-gray-600 w-full"
        data-testid="progress-bar"
      >
        <div
          class="h-full bg-primary-500"
          style="width: 0%;"
        />
      </div>
      <div
        class="flex items-center p-2"
      >
        <button
          aria-label="Play"
          class="mr-2"
        >
          ▶
        </button>
        <button
          aria-label="Pause"
          class="mr-2"
        >
          ⏸
        </button>
        <button
          aria-label="Mute"
          class="mr-2"
        >
          🔇
        </button>
        <button
          aria-label="Unmute"
          class="mr-2 hidden"
        >
          🔊
        </button>
        <input
          aria-label="Volume"
          class="w-24 mr-2"
          max="1"
          min="0"
          step="0.1"
          type="range"
          value="1"
        />
        <button
          aria-label="Enter fullscreen"
        >
          ⛶
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MediaPlayer Snapshots renders with loop correctly 1`] = `
<DocumentFragment>
  <div
    class="relative "
    data-testid="media-player"
  >
    <video
      class="w-full h-full"
      controls=""
      data-testid="video-element"
      loop=""
      src="test-video.mp4"
    />
    <div
      class="absolute bottom-0 left-0 right-0"
      data-testid="media-controls"
    >
      <div
        class="h-1 bg-gray-600 w-full"
        data-testid="progress-bar"
      >
        <div
          class="h-full bg-primary-500"
          style="width: 0%;"
        />
      </div>
      <div
        class="flex items-center p-2"
      >
        <button
          aria-label="Play"
          class="mr-2"
        >
          ▶
        </button>
        <button
          aria-label="Pause"
          class="mr-2"
        >
          ⏸
        </button>
        <button
          aria-label="Mute"
          class="mr-2"
        >
          🔇
        </button>
        <button
          aria-label="Unmute"
          class="mr-2 hidden"
        >
          🔊
        </button>
        <input
          aria-label="Volume"
          class="w-24 mr-2"
          max="1"
          min="0"
          step="0.1"
          type="range"
          value="1"
        />
        <button
          aria-label="Enter fullscreen"
        >
          ⛶
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MediaPlayer Snapshots renders with multiple sources correctly 1`] = `
<DocumentFragment>
  <div
    class="relative "
    data-testid="media-player"
  >
    <video
      class="w-full h-full"
      controls=""
      data-testid="video-element"
    >
      <source
        data-testid="source-element"
        src="test-video.mp4"
        type="video/mp4"
      />
      <source
        data-testid="source-element"
        src="test-video.webm"
        type="video/webm"
      />
    </video>
    <div
      class="absolute bottom-0 left-0 right-0"
      data-testid="media-controls"
    >
      <div
        class="h-1 bg-gray-600 w-full"
        data-testid="progress-bar"
      >
        <div
          class="h-full bg-primary-500"
          style="width: 0%;"
        />
      </div>
      <div
        class="flex items-center p-2"
      >
        <button
          aria-label="Play"
          class="mr-2"
        >
          ▶
        </button>
        <button
          aria-label="Pause"
          class="mr-2"
        >
          ⏸
        </button>
        <button
          aria-label="Mute"
          class="mr-2"
        >
          🔇
        </button>
        <button
          aria-label="Unmute"
          class="mr-2 hidden"
        >
          🔊
        </button>
        <input
          aria-label="Volume"
          class="w-24 mr-2"
          max="1"
          min="0"
          step="0.1"
          type="range"
          value="1"
        />
        <button
          aria-label="Enter fullscreen"
        >
          ⛶
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MediaPlayer Snapshots renders with playback speed controls correctly 1`] = `
<DocumentFragment>
  <div
    class="relative "
    data-testid="media-player"
  >
    <video
      class="w-full h-full"
      controls=""
      data-testid="video-element"
      src="test-video.mp4"
    />
    <div
      class="absolute bottom-0 left-0 right-0"
      data-testid="media-controls"
    >
      <div
        class="h-1 bg-gray-600 w-full"
        data-testid="progress-bar"
      >
        <div
          class="h-full bg-primary-500"
          style="width: 0%;"
        />
      </div>
      <div
        class="flex items-center p-2"
      >
        <button
          aria-label="Play"
          class="mr-2"
        >
          ▶
        </button>
        <button
          aria-label="Pause"
          class="mr-2"
        >
          ⏸
        </button>
        <button
          aria-label="Mute"
          class="mr-2"
        >
          🔇
        </button>
        <button
          aria-label="Unmute"
          class="mr-2 hidden"
        >
          🔊
        </button>
        <input
          aria-label="Volume"
          class="w-24 mr-2"
          max="1"
          min="0"
          step="0.1"
          type="range"
          value="1"
        />
        <button
          aria-label="Enter fullscreen"
        >
          ⛶
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MediaPlayer Snapshots renders with poster image correctly 1`] = `
<DocumentFragment>
  <div
    class="relative "
    data-testid="media-player"
  >
    <video
      class="w-full h-full"
      controls=""
      data-testid="video-element"
      poster="poster.jpg"
      src="test-video.mp4"
    />
    <div
      class="absolute bottom-0 left-0 right-0"
      data-testid="media-controls"
    >
      <div
        class="h-1 bg-gray-600 w-full"
        data-testid="progress-bar"
      >
        <div
          class="h-full bg-primary-500"
          style="width: 0%;"
        />
      </div>
      <div
        class="flex items-center p-2"
      >
        <button
          aria-label="Play"
          class="mr-2"
        >
          ▶
        </button>
        <button
          aria-label="Pause"
          class="mr-2"
        >
          ⏸
        </button>
        <button
          aria-label="Mute"
          class="mr-2"
        >
          🔇
        </button>
        <button
          aria-label="Unmute"
          class="mr-2 hidden"
        >
          🔊
        </button>
        <input
          aria-label="Volume"
          class="w-24 mr-2"
          max="1"
          min="0"
          step="0.1"
          type="range"
          value="1"
        />
        <button
          aria-label="Enter fullscreen"
        >
          ⛶
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`MediaPlayer Snapshots renders with tracks correctly 1`] = `
<DocumentFragment>
  <div
    class="relative "
    data-testid="media-player"
  >
    <video
      class="w-full h-full"
      controls=""
      data-testid="video-element"
      src="test-video.mp4"
    >
      <track
        data-testid="track-element"
        kind="subtitles"
        label="English"
        src="subtitles-en.vtt"
        srclang="en"
      />
      <track
        data-testid="track-element"
        kind="subtitles"
        label="Deutsch"
        src="subtitles-de.vtt"
        srclang="de"
      />
    </video>
    <div
      class="absolute bottom-0 left-0 right-0"
      data-testid="media-controls"
    >
      <div
        class="h-1 bg-gray-600 w-full"
        data-testid="progress-bar"
      >
        <div
          class="h-full bg-primary-500"
          style="width: 0%;"
        />
      </div>
      <div
        class="flex items-center p-2"
      >
        <button
          aria-label="Play"
          class="mr-2"
        >
          ▶
        </button>
        <button
          aria-label="Pause"
          class="mr-2"
        >
          ⏸
        </button>
        <button
          aria-label="Mute"
          class="mr-2"
        >
          🔇
        </button>
        <button
          aria-label="Unmute"
          class="mr-2 hidden"
        >
          🔊
        </button>
        <input
          aria-label="Volume"
          class="w-24 mr-2"
          max="1"
          min="0"
          step="0.1"
          type="range"
          value="1"
        />
        <button
          aria-label="Enter fullscreen"
        >
          ⛶
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
